File: 设计模式/程序/23-TemplatePattern/TemplatePattern(C++)/src/CricketGame.cpp
#include "CricketGame.h"
#include <iostream>


namespace TemplatePatternApp
{
    bool CricketGame::Initialize() const 
    {
        std::cout << "Wait Players...\n";
        std::cout << "Prepare Cricket Ground...\n";
        std::cout << "Get Ball\n...";
        return true;
    }
    int CricketGame::PlayGame() const 
    {
        std::cout << "Playing...\n";
        std::cout << "hithithit\n";
        return 11;
    }
    void CricketGame::EndGame() const 
    {
        std::cout << "Clean Cricket Ground...\n";
    }
}

File: 设计模式/程序/23-TemplatePattern/TemplatePattern(python)/CricketGame.py
from Game import Game 

class CricketGame(Game):
    # 实现三个函数
    def Initial(self):
        print('Prepare Cricket Ground...')
        print('Wait Players...')
        print('Wait Ball...')
        return True

    def PlayGame(self):
        print('Playing...')
        print('hithithit')
        return 12

    def EndGame(self):
        print('Player Take Shower...')
        print('Clean Cricket ground...')
        print('Customer leaves...')